src/main/fc/init.c

```diff
@@ -800,15 +800,20 @@ void init(void)
 
     if (featureIsEnabled(FEATURE_OSD)) {
 #if defined(USE_MAX7456)
-        // If there is a max7456 chip for the OSD then use it
+        // If there is a max7456 chip for the OSD configured and detectd then use it.
         osdDisplayPort = max7456DisplayPortInit(vcdProfile());
-#elif defined(USE_CMS) && defined(USE_MSP_DISPLAYPORT) && defined(USE_OSD_OVER_MSP_DISPLAYPORT) // OSD over MSP; not supported (yet)
-        osdDisplayPort = displayPortMspInit();
 #endif
-        // osdInit  will register with CMS by itself.
+
+#if defined(USE_CMS) && defined(USE_MSP_DISPLAYPORT) && defined(USE_OSD_OVER_MSP_DISPLAYPORT)
+        if (!osdDisplayPort) {
+            osdDisplayPort = displayPortMspInit();
+        }
+#endif
```

Comment:
I'm not sure about this block. `displayPortMspInit()` always returns a pointer to a `displayPort_t`. So anytime we fall through a failed max7456 detection we're going to end up with the msp displayPort. Then OSD initialization succeeds and the OSD task continues to run.

Reply:
Yes, this is the case - this is identical to 4.2 behaviour if `osd_displayport_device` is set to `auto` (the default). Since we are not running into timeouts trying to push data over SPI to a MAX7456 that isn't responding keeping the OSD task running should not be a problem.
To me this seemed the most useful behaviour, as it will allow the use of MSP based OSD devices on Unified Targets, just like they were usable on legacy targets without an OSD chip.
Do you think this should be different?

Reply:
Also, if a board does not have an OSD chip and MSP OSD is not used then the OSD feature should be disabled, and this will keep the OSD task from running:

```
# diff

# version
# Betaflight / STM32F405 (S405) 4.1.3 Jan 14 2020 / 19:38:52 (9b1b4844d2) MSP API: 1.42
# manufacturer_id: AIRB   board_name: OMNIBUSF4   custom defaults: YES

# start the command batch
batch start

board_name OMNIBUSF4
manufacturer_id AIRB

# feature
feature -OSD

# master
set max7456_spi_bus = 2

profile 0

rateprofile 0

# end the command batch
batch end

# tasks
Task list             rate/hz  max/us  avg/us maxload avgload  total/ms
00 - (         SYSTEM)      9       6       1    0.5%    0.5%        12
01 - (         SYSTEM)    985       7       1    1.1%    0.5%      1504
02 - (            PID)   1965      69      27   14.0%    5.8%    467404
   - (           GYRO)   7863
03 - (            ACC)    985      16       9    2.0%    1.3%     18148
04 - (       ATTITUDE)     99      15       8    0.6%    0.5%      1789
05 - (             RX)     32      49      26    0.6%    0.5%      1894
06 - (         SERIAL)     99   46692       4  462.7%    0.5%      4821
07 - (       DISPATCH)    985       7       0    1.1%    0.0%      1204
08 - (BATTERY_VOLTAGE)     49       7       1    0.5%    0.5%       143
09 - (BATTERY_CURRENT)     49       7       0    0.5%    0.0%        64
10 - ( BATTERY_ALERTS)      4       7       2    0.5%    0.5%        16
11 - (         BEEPER)     99       8       1    0.5%    0.5%       199
23 - (            CMS)     59       8       1    0.5%    0.5%       115
24 - (        VTXCTRL)      4       6       0    0.5%    0.0%         7
25 - (        CAMCTRL)      4       3       1    0.5%    0.5%         5
27 - (    ADCINTERNAL)      0       4       2    0.5%    0.5%         4
28 - (       PINIOBOX)     19       7       1    0.5%    0.5%        33
RX Check Function                   8       1                       124
Total (excluding SERIAL)                        24.4%   12.6%
```